porebuilder/tests/test_porebuilder.py

```diff
@@ -50,3 +53,35 @@ def test_particles_in_box(self, GraphenePoreSolvent):
             assert particle.xyz[0][0] < box.maxs[0]
             assert particle.xyz[0][1] < box.maxs[1]
             assert particle.xyz[0][2] < box.maxs[2]
+
+    def test_functionalize_surface(self):
+
+        class H(mb.Compound):
+            def __init__(self):
+                super(H, self).__init__()
+
+                self.add(mb.Particle(name='H'))
+                up_port = mb.Port(
+                    anchor=self[0], orientation=[
+                        0, 1, 0], separation=.075)
+                self.add(up_port, "up")
+
+        class O(mb.Compound):
+            def __init__(self):
+                super(O, self).__init__()
+
+                self.add(mb.Particle(name='O'))
+                up_port = mb.Port(
+                    anchor=self[0], orientation=[
+                        0, 1, 0], separation=.075)
+                self.add(up_port, "up")
+
+        with pytest.raises(ValueError):
+            pb.GraphenePoreFunctionalized(
+                func_groups=[H(), O()], func_percent=[.03, .04, .03])
+
+        for per in range(1, 10):
+            pore = pb.GraphenePoreFunctionalized(
+                func_groups=H(), func_percent=per / 10)
+            assert(pore.n_particles - 2688 > (per - .3) / 10 *
```

Comment:
Let's split this into two `assert` statements. And I'm not sure where 864 comes from?

Reply:
864 is the total number of inner surface carbons that could potentially be fictionalized. Now that I changed the function so that it adds exactly the right amount of new groups instead of relying on dice rolls I'll be re-writing the test cases next.